tchannel-core/src/main/java/com/uber/tchannel/api/TChannel.java

```diff
@@ -134,6 +131,10 @@ public long getInitTimeout() {
         return this.initTimeout;
     }
 
+    public boolean hasListened() {
+        return !listeningHost.equals("0.0.0.0");
```

Comment:
if TChannel.listen() is not called, the default value for listeningHost would be "0.0.0.0"